Nasal/light-manager.nas

```diff
@@ -0,0 +1,303 @@
+
+        apos.set_lat(lat + ((me.light1_xpos + proj_x) * ch + me.light1_ypos * sh) / me.lat_to_m);
+        apos.set_lon(lon + ((me.light1_xpos + proj_x)* sh - me.light1_ypos * ch) / me.lon_to_m);
+
+        var delta_x = (apos.lat() - vpos.lat()) * me.lat_to_m;
+        var delta_y = -(apos.lon() - vpos.lon()) * me.lon_to_m;
+        var delta_z = apos.alt()- proj_z - vpos.alt();
+
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-x-m", delta_x);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-y-m", delta_y);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-z-m", delta_z);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/dir", heading);
+
+        # light 2 position
+        var alt_agl = getprop("/position/altitude-agl-ft");
+
+        var proj_x = alt_agl;
+        var proj_z = alt_agl/10.0;
+
+        apos.set_lat(lat + ((me.light2_xpos + proj_x) * ch + me.light2_ypos * sh) / me.lat_to_m);
+        apos.set_lon(lon + ((me.light2_xpos + proj_x)* sh - me.light2_ypos * ch) / me.lon_to_m);
+
+        var delta_x = (apos.lat() - vpos.lat()) * me.lat_to_m;
+        var delta_y = -(apos.lon() - vpos.lon()) * me.lon_to_m;
+        var delta_z = apos.alt()- proj_z - vpos.alt();
+
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-x-m[1]", delta_x);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-y-m[1]", delta_y);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-z-m[1]", delta_z);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/dir", heading);
+
+        # light 3 position
+        apos.set_lat(lat + (me.light2_xpos * ch + me.light3_ypos * sh) / me.lat_to_m);
+        apos.set_lon(lon + (me.light2_xpos * sh - me.light3_ypos * ch) / me.lon_to_m);
+
+        delta_x = (apos.lat() - vpos.lat()) * me.lat_to_m;
+        delta_y = -(apos.lon() - vpos.lon()) * me.lon_to_m;
+        delta_z = apos.alt() - vpos.alt();
+
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-x-m[2]", delta_x);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-y-m[2]", delta_y);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-z-m[2]", delta_z);
+
+        # light 4 position
+        apos.set_lat(lat + (me.light3_xpos * ch + me.light4_ypos * sh) / me.lat_to_m);
+        apos.set_lon(lon + (me.light3_xpos * sh - me.light4_ypos * ch) / me.lon_to_m);
+
+        delta_x = (apos.lat() - vpos.lat()) * me.lat_to_m;
+        delta_y = -(apos.lon() - vpos.lon()) * me.lon_to_m;
+        delta_z = apos.alt() - vpos.alt();
+
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-x-m[3]", delta_x);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-y-m[3]", delta_y);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-z-m[3]", delta_z);
+
+        settimer(func me.update(), 0.0);
+    },   
+
+    light1_on: func {
```

Comment:
When I look at all these lightx_on/off functions, they look very similar. You can simplify the code by creating a function that returns a function for you. Then you have only 1 place where you actually set the lightspot values and where you define the ALS properties:

```javascript
var lightspot_on_func = func (index) {
    return func {
        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r", index, me.light_r[index]);
        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g", index, me.light_g[index]);
        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b", index, me.light_b[index]);
    }
}

var light1_on = lightspot_on_func(0);
```

Just make sure that `me.light_r`, etc. are arrays.

You can do the same thing for `lightspot_off_func` and for the other lights.